bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+    def __init__(self, token, filter):
+        self._api = TodoistAPI(token)
+        self.filter = filter
+
+    def get_projects(self):
+        all_projects = []
+        projects_iter = self._api.get_projects()
+        for projects in projects_iter:
+            for project in projects:
+                all_projects.append(project)
+        return all_projects
+
+    def get_sections(self):
+        all_sections = []
+        sections_iter = self._api.get_sections()
+        for sections in sections_iter:
+            for section in sections:
+                all_sections.append(section)
+        return all_sections
+
+    def get_users(self, project_id):
+        all_users = []
+        users_iter = self._api.get_collaborators(project_id)
+        for users in users_iter:
+            for user in users:
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        return tasks_iter
+
+
+class TodoistIssue(Issue):
+    ASSIGNEE = "todoistassignee"
+    ASSIGNER = "todoistassigner"
+    CONTENT = "todoistcontent"
+    DESCRIPTION = "todoistdescription"
+    DURATION = "todoistduration"
+    ID = "todoistid"
+    SECTION = "todoistsection"
+    URL = "todoisturl"
+
+    PRIORITY_MAP = {
+        4: "H",
+        3: "M",
+        2: "L",
+        1: None,
+    }
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Todoist ID",
+        },
+        CONTENT: {
+            "type": "string",
+            "label": "Todoist Content",
+        },
+        DESCRIPTION: {
+            "type": "string",
+            "label": "Todoist Description",
+        },
+        DURATION: {
+            "type": "string",
+            "label": "Todoist Duration",
+        },
+        SECTION: {
+            "type": "string",
+            "label": "Todoist Section",
+        },
+        ASSIGNEE: {
+            "type": "string",
+            "label": "Todoist Assignee",
+        },
+        ASSIGNER: {
+            "type": "string",
+            "label": "Todoist Assigner",
+        },
+        URL: {
+            "type": "string",
+            "label": "Todoist URL",
+        },
+    }
+
+    UNIQUE_KEY = (ID, ID)
+
+    # replace characters that cause escaping issues in teh description like [] and "
+    # this is a workaround for https://github.com/ralphbean/taskw/issues/172
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    def to_taskwarrior(self):
+        default_time = time(0, 0, 0)
+        # use due date "scheduled".
+        # adjust timezone to use local time for "floating" dates
+        if self.record.due and type(self.record.due.date) is datetime:
```

Comment:
This is what I'd typically expect to see here. The functional difference is that it will be True if the date is a subclass of datetime but I'd think that would be ok here.
```suggestion
        if self.record.due and isinstance(self.record.due.date, datetime):
```

Reply:
The Todosit API is returning either a `date` or a `datetime`.  The check here is specifically to see if it's a `datetime`

Reply:
Ok, but a `date` is not a `datetime`:

```py
[ins] In [1]: import datetime

[ins] In [2]: today = datetime.date.today()

[ins] In [3]: isinstance(today, datetime.datetime)
Out[3]: False

[ins] In [4]: isinstance(today, datetime.date)
Out[4]: True
```

Reply:
switched to use `isinstance()`